database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1576,137 +1896,57 @@ CREATE TABLE IF NOT EXISTS application_list_shape (
-        LEFT JOIN public.application_stage_status_latest AS stage_status ON app.id = stage_status.application_id
-        LEFT JOIN public.organisation org ON app.org_id = org.id
+        app.reviewer_list AS reviewers,
+        app.assigner_list AS assigners,
+        reviewer_action,
+        assigner_action
+        FROM application app
+        LEFT JOIN application_stage_status_latest as stage_status
+            ON app.id = stage_status.application_id
+        LEFT JOIN "user" 
+            ON app.user_id = "user".id
+        LEFT JOIN organisation
+            ON app.org_id = organisation.id
         LEFT JOIN trigger_schedule ts ON app.id = ts.application_id
-            AND ts.is_active = TRUE
-            AND ts.event_code = 'applicantDeadline'
-            -- REVIEWERS and ASSIGNERS
-    LEFT JOIN review_assignment AS assigned_assignment ON (assigned_assignment.application_id = app.id
-            AND assigned_assignment.stage_id = stage_status.stage_id
-            AND assigned_assignment.status = 'ASSIGNED')
-    LEFT JOIN public."user" AS assigner_user ON assigned_assignment.assigner_id = assigner_user.id
-    LEFT JOIN public."user" AS reviewer_user ON assigned_assignment.reviewer_id = reviewer_user.id
-    -- REVIEWER ACTIONS
-    LEFT JOIN review_assignment AS reviewer_assignment ON (reviewer_assignment.application_id = app.id
-            AND reviewer_assignment.stage_id = stage_status.stage_id
-            AND app.outcome = 'PENDING'
-            AND reviewer_assignment.reviewer_id = userId)
-    LEFT JOIN review AS action_review ON action_review.review_assignment_id = reviewer_assignment.id
-    LEFT JOIN review_status_history AS action_review_status_history ON (action_review_status_history.review_id = action_review.id
-            AND action_review_status_history.is_current = TRUE)
-        -- ASSIGNER ACTIONS
-    LEFT JOIN assigned_sections_by_stage_and_level ON (assigned_sections_by_stage_and_level.application_id = app.id
-            AND assigned_sections_by_stage_and_level.stage_id = stage_status.stage_id
-            AND app.outcome = 'PENDING'
-            AND assigned_sections_by_stage_and_level.assigner_id = userId)
-    LEFT JOIN template_section AS assignable_sections ON assignable_sections.template_id = template.id
-    LEFT JOIN allowed_self_assignable_sections(userId) AS allowed_self_assignable_sections ON allowed_self_assignable_sections.review_assignment_id = reviewer_assignment.id
-    LEFT JOIN review_assignment_assigned_sections(userId) as review_assignment_assigned_sections ON review_assignment_assigned_sections.review_assignment_id = reviewer_assignment.id
-WHERE
-    app.is_config = FALSE
-GROUP BY
-    1,
-    2,
-    3,
-    4,
-    5,
-    6,
-    7,
-    8,
-    9,
-    10,
-    11,
-    12,
-    13,
-    stage_status.stage_id
-
-$$
-LANGUAGE sql
-STABLE;
+                AND ts.is_active = TRUE
+                AND ts.event_code = 'applicantDeadline'
+        LEFT JOIN (
```

Comment:
in the other left joins we filter out in 'on' statement, not sure about the performance implications but it seems more directr to

```sql
LEFT JOIN application_reviewer_action as ara ON ara.user_id = user_id AND ara.id = app.application_id 
```

Reply:
I think I did this intentionally, as the "application_reviewer_action" table can get huge, and we'll only ever be joining to a very small selection of its lines at any time. However, I originally had it just joining on `ara.user_id = user_id` and putting a WHERE at the end to filter it by application, which seemed really inefficient to me. 

But I didn't think of doing a two-condition Join like you suggested, so I guess that would probably be basically the same. Have updated: d5cd051f

Reply:
🙏